ktxspec.adoc

```diff
@@ -562,36 +563,11 @@ Refer to <<_texture_type>> for more details about valid values.
 === faceCount
 `faceCount` specifies the number of cubemap faces. For cubemaps and
 cubemap arrays this must be 6. For non cubemaps this must be 1.
-Cubemap faces are stored in the order: +X, -X, +Y, -Y, +Z, -Z.
-
-As prescribed by the Vulkan and OpenGL specifications, all faces
-must have a right-down orientation, i.e., an upper left origin, and
-must be arranged using a left-handed coordinate system with +Y up.
-For example in an outdoor scene, the +Y face must be the sky.  When
-facing in the scene what is shown in the +Z face, what is on the
-left must be shown in the -X face and what is on the right in the
-+X face.  See the identical face selection tables and the equations
-for calculating (s, t) in
-{url-khr-vulkan}/specs/1.2-extensions/html/vkspec.html#_cube_map_face_selection_and_transformations[section
-15.5.3 Cube Map Face Selection and Transformations] of <<VULKAN12>>
-and
-{url-khr-reg}/OpenGL/specs/gl/glspec46.core.pdf#nameddest=section-8.13[section
-8.13 Cube Map Texture Selection] of <<OPENGL46>>.
-
-[TIP]
-====
-OpenGL has a right-handed coordinate system. Therefore when using a
-skybox cube's coordinates as the uvw coordinates for sampling a cubemap
-those coordinates must first be transformed to a left-handed system by
-multiplying the Z coordinate by -1. Failure to do this will result in
-the displayed scene being a mirror image of reality, a common error in
-samples found on the web.
-
-While Vulkan has a left-handed coordinate system, +Y is down, therefore
-skybox coordinates must be rotated 180 degrees around the X axis before
-they are used to sample a cubemap. Failure to do this will result in the
-scene being upside down.
-====
+Cubemap faces are stored in the order: +X, -X, +Y, -Y, +Z, -Z in a
+left-handed coordinate system with +Y up. All faces must have the
```

Comment:
Slightly opinionated: it's not enough to define +Y as "up" to set the "ground truth" for a cubemap that is supposed to represent a real environment. One more axis needs to be tied (front/back/left/right).

Reply:
Please suggest modified language to address this. I don't have time for to and fro. I'm supposed to produce the release version of the spec. today and it is already 8:30PM.

Reply:
On it. Will check with glTF's axises.

Reply:
glTF defines its object space as: right-handed, **+Y up**, **+Z forward**, **+X left**.

Given that KTX cubemaps are left-handed, we have two options:
- +Z forward, +X right. In that case, transformation would be multiplying X by -1.
- -Z forward, +X left. In that case, transformation would be multiplying Z by -1.

Reply:
My cubemap samples have +Z forward and +X on the right. However I get the correct result in OpenGL by multiplying the cube coordinates' Z by -1. In Vulkan I get the correct result by rotating the cube's coordinate system around the X axis to because Vulkan has +Y down.

Wait a minute. What coordinates are you talking about? Are these cubemap face coordinates or skybox cube coordinates? I'm talking about the face coordinates.

Reply:
And if we're talking face coordinates there is no difference between your 2 cases.

Reply:
In my opinion, the first option is better as it seems to be more popular across tools.

Reply:
+X left, seriously?

Reply:
My samples are using the GL RH-coord system. That must be why I have to multiply Z by -1.

What language do I need to modify? I think I just need to say +Z forward, +X on the right. The necessary transformation depends on the API or glTF or whatever. Is that correct?

Reply:
What is "forward" here? Does it mean in front of you so  going into the screen, i.e, -Z direction in OpenGL default CS? So glTF has +Z into the screen.

Reply:
@lexaknyazev please reply. I need to wrap this up.

Reply:
> What coordinates are you talking about?

From the point of view of an observer that is located inside the cube, +Y is up, +X is on the right, and +Z is forward. That's the proposal. Alternative is to make -X right and -Z forward.

Reply:
For the reference, here's glTF's object space:
![](https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/figures/coordinate-system.png?raw=true)

Reply:
> I think I just need to say +Z forward, +X on the right. 

Yeah, that's enough.

Reply:
Inside the cube. That's what I thought you were talking aboutl. The object space you picture is the same as OpenGL.

It seems to me the 2 alternatives are the same. If you are looking at the posz image and you have the posx image on your right, if you turn around so you are looking at the negz image the posx image will now be on your left.

So why is my sample working when I multiply z by -1? 

I have to give up tonight. I can't think properly any more.

As I asked earlier please propose modifications to the existing language.